include/network_interface/network_utils.h

```diff
@@ -24,6 +24,16 @@ namespace Network
     LE
   };
 
+  	inline int system_is_big_endian(uint32_t input)
+	{
+		union {
+			uint32_t i;
+			char c[4];
+		} big_int = {input};
+
+		return big_int.c[0] == 1; 
+	}
+
```

Comment:
Would you mind implementing it as:

```bool is_big_endian()
{
    union {
        uint32_t i;
        char c[4];
    } bint = {0x01020304};

    return bint.c[0] == 1; 
}```

This one doesn't require making a copy of the read value each time and guarantees that the value being checked won't have 1s in the highest bit of both the LSB and the MSB (which would defeat the version you have above). This is a C++ version of the one at [https://stackoverflow.com/questions/1001307/detecting-endianness-programmatically-in-a-c-program](https://stackoverflow.com/questions/1001307/detecting-endianness-programmatically-in-a-c-program).

Reply:
yep, good call, commit on its way